### PR TITLE
added data removal request

### PR DESCRIPTION
**Pull request by ewood-ac**
Added data removal request code.

### CODE REVIEW

**Review by pgraham3 — APPROVED** [review]
👍 Thanks!

**Review by balvig — APPROVED** [review]
👏 👍  I'm not particular familiar with the PHP lib so not sure what we usually do for tests.

I can see that there are some in the `/tests` folder and we usually do add them to the other languages (see https://github.com/ActiveCampaign/postmark-gem/pull/147 f.ex), but will leave it to your discretion, as its been many many years since I last worked with PHP! 😅  🙇

File: composer.json [hunk 1]
```diff
@@ -7,11 +7,11 @@
     "license": "MIT",
     "description": "The officially supported client for Postmark (http://postmarkapp.com)",
     "require": {
-        "php": ">=7.0.0",
+        "php": ">=8",
```
Comment (gsteel) [inline composer.json:10]: Can you be sure that this lib will run without any problems on PHP 8.3, or 9.0, or 10.0? Surely this should be `~8.0 || ~8.1 || ~8.2`

File: .circleci/config.yml [hunk 2]
```diff
@@ -6,28 +6,14 @@ workflows:
...
-          version: "7.3"
-          requires:
-            - php72
-      - unit-tests:
-          name: php74
-          version: "7.4"
-          requires:
-            - php73
+          name: php82
+          version: "8.2"
```
Comment (gsteel) [inline .circleci/config.yml:16]: 8.3 is getting released in November - it's widely available in RC state in most distros.

File: tests/PostmarkAdminClientDataRemovalTest.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        $createRequest = $client->createDataRemovalRequest("test@activecampaign.com", "test2@activecampaign.com", true);
+        $checkRequest = $client->getDataRemoval($createRequest->ID);
+
+        $this->assertNotEmpty($checkRequest->ID);
+        $this->assertNotEmpty($checkRequest->Status);
+        $this->assertEquals($checkRequest->ID, $createRequest->ID);
+    }
+}
+
+?>
```
Comment (gsteel) [inline tests/PostmarkAdminClientDataRemovalTest.php:34]: PHP files should not contain closing tags

**Review by gsteel — COMMENTED** [review]
Apart from desperately needing a coding standard, why the major release for a feature addition?

File: composer.json (hunk 1 above)
Comment (ewood-ac) [reply composer.json:10]: Thanks I will update this.

**Pull request merged**